src/components/MapGL/index.js

```diff
@@ -443,6 +444,37 @@ class MapGL extends PureComponent<Props, State> {
     return this._map;
   }
 
+  /**
+   * Update Map event listeners from newProps
+   *
+   * @private
+   * @param {Props} prevProps
+   * @param {Props} newProps
+   */
+  _updateMapEventListeners(prevProps: Props, newProps: Props): void {
+    events.forEach((event) => {
+      const propName = `on${capitalizeFirstLetter(event)}`;
+      const prevEventHandler = prevProps[propName];
+      const newEventHandler = newProps[propName];
+      if (prevEventHandler !== newEventHandler) {
```

Comment:
Could you please use `isArraysEqual` for comparing these props? https://github.com/urbica/react-map-gl/blob/12906655d76256731d4020cbc3883a7996c7463e/src/utils/isArraysEqual.js#L1

Reply:
I'm guessing something like this?

```js
if (!(prevEventHandler === newEventHandler ||                                                                                                                                                                                                                                                                   
    (Array.isArray(prevEventHandler) &&
     Array.isArray(newEventHandler) &&
     isArraysEqual(prevEventHandler, newEventHandler)))) {
```

No worries on the delay!

Reply:
Yep, but with array comparison first 👍

Reply:
Done!